hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/ListingBasedRollbackStrategy.java

```diff
@@ -317,56 +293,45 @@ private static String formatDeletePath(String path) {
+  private List<StoragePath> fetchFilesFromInstant(HoodieInstant instantToRollback,
+                                                  String partitionPath, String basePath,
+                                                  String baseFileExtension, HoodieStorage storage,
+                                                  Option<HoodieCommitMetadata> commitMetadataOptional,
+                                                  boolean isCommitMetadataCompleted,
+                                                  HoodieTableType tableType,
+                                                  HoodieTableVersion tableVersion) throws IOException {
+    // for MOR tables with version < 8, listing is required to fetch the log files associated with base files added by this commit.
+    if (isCommitMetadataCompleted && (tableType == HoodieTableType.COPY_ON_WRITE || tableVersion.greaterThanOrEquals(HoodieTableVersion.EIGHT))) {
+      return fetchFilesFromCommitMetadata(instantToRollback, partitionPath, basePath, commitMetadataOptional.get(), baseFileExtension);
     } else {
       return fetchFilesFromListFiles(instantToRollback, partitionPath, basePath, baseFileExtension, storage);
     }
   }
 
-  private List<StoragePathInfo> fetchFilesFromCommitMetadata(HoodieInstant instantToRollback,
-                                                             String partitionPath,
-                                                             String basePath,
-                                                             HoodieCommitMetadata commitMetadata,
-                                                             String baseFileExtension,
-                                                             HoodieStorage storage) throws IOException {
+  private List<StoragePath> fetchFilesFromCommitMetadata(HoodieInstant instantToRollback,
+                                                         String partitionPath,
+                                                         String basePath,
+                                                         HoodieCommitMetadata commitMetadata,
+                                                         String baseFileExtension) {
     StoragePathFilter pathFilter = getPathFilter(baseFileExtension,
         instantToRollback.requestedTime());
-    List<StoragePath> filePaths = getFilesFromCommitMetadata(basePath, commitMetadata, partitionPath)
-        .filter(entry -> {
-          try {
-            return storage.exists(entry);
-          } catch (Exception e) {
-            LOG.error("Exists check failed for " + entry.toString(), e);
-          }
-          // if any Exception is thrown, do not ignore. let's try to add the file of interest to be deleted. we can't miss any files to be rolled back.
-          return true;
-        }).collect(Collectors.toList());
-
-    return storage.listDirectEntries(filePaths, pathFilter);
+    return getFilesFromCommitMetadata(basePath, commitMetadata, partitionPath)
```

Comment:
The file existence check seems unnecessary because we can delete a file that does not exist, cc @nsivabalan for the background of this check.